File: src/test/java/no/nav/pensjon/selvbetjeningopptjening/common/domain/PersonTest.java
package no.nav.pensjon.selvbetjeningopptjening.common.domain;

import no.nav.pensjon.selvbetjeningopptjening.PidGenerator;
import org.junit.jupiter.api.Test;

import java.time.LocalDate;

import static org.junit.jupiter.api.Assertions.*;

class PersonTest {

    @Test
    void should_use_fodselsdato_when_fodselsdato_argument_not_null(){
        LocalDate expectedFodselsdato = LocalDate.now().minusYears(65);
        Person person = new Person(PidGenerator.generatePidAtAge(65), null, null, null, new BirthDate(expectedFodselsdato));

        assertEquals(expectedFodselsdato, person.getFodselsdato());
    }

    @Test
    void should_use_default_fodselsdato_from_pid_when_fodselsdato_argument_is_null(){
        LocalDate expectedFodselsdato = LocalDate.now().minusYears(65);
        Person person = new Person(PidGenerator.generatePid(expectedFodselsdato), null, null, null, null);

        assertEquals(expectedFodselsdato, person.getFodselsdato());
    }

    @Test
    void should_set_names_on_person(){
        String expectedFornavn = "fornavn";
        String expectedMellomnavn = "mellomnavn";
        String expectedEtternavn = "etternavn";

        Person person = new Person(PidGenerator.generatePidAtAge(65), expectedFornavn, expectedMellomnavn, expectedEtternavn, null);

        assertEquals(expectedFornavn, person.getFornavn());
        assertEquals(expectedMellomnavn, person.getMellomnavn());
        assertEquals(expectedEtternavn, person.getEtternavn());
    }

}

File: src/test/java/no/nav/pensjon/selvbetjeningopptjening/person/PersonTest.kt
package no.nav.pensjon.selvbetjeningopptjening.person

import io.kotest.core.spec.style.ShouldSpec
import io.kotest.matchers.shouldBe
import no.nav.pensjon.selvbetjeningopptjening.PidGenerator
import java.time.LocalDate

class PersonTest : ShouldSpec({

    should("use fødselsdato when fødselsdato argument not null") {
        val expectedFoedselsdato = LocalDate.now().minusYears(65)

        Person(
            pid = PidGenerator.generatePidAtAge(65),
            fornavn = null,
            mellomnavn = null,
            etternavn = null,
            foedselsdato = Foedselsdato2(expectedFoedselsdato)
        ).getFodselsdato() shouldBe expectedFoedselsdato
    }

    should("set names on person") {
        val expectedFornavn = "fornavn"
        val expectedMellomnavn = "mellomnavn"
        val expectedEtternavn = "etternavn"

        with(
            Person(
                pid = PidGenerator.generatePidAtAge(65),
                fornavn = expectedFornavn,
                mellomnavn = expectedMellomnavn,
                etternavn = expectedEtternavn,
                foedselsdato = null
            )
        ) {
            fornavn shouldBe expectedFornavn
            mellomnavn shouldBe expectedMellomnavn
            etternavn shouldBe expectedEtternavn
        }
    }
})
